Challenges/Day_2/backup_with_rotation.sh

```
@@ -1,46 +1,52 @@
-# Function to perform rotation and keep only the last 3 backups
 function perform_rotation {
-    local backups=($(ls -t "${source_dir}/backup_"*.zip 2>/dev/null))
-
-    if [ "${#backups[@]}" -gt 3 ]; then
-        local backups_to_remove=("${backups[@]:3}")
-        for backup in "${backups_to_remove[@]}"; do
-            rm -f "$backup"
-        done
-    fi
+
+        backups=($(ls -t "${backup_dir}/backup_"*.zip 2>/dev/null))
```

Shellcheck

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)